src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/RazorRenameProvider.ts

```diff
@@ -0,0 +1,51 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as vscode from 'vscode';
+import { RazorDocumentManager } from './RazorDocumentManager';
+import { RazorDocumentSynchronizer } from './RazorDocumentSynchronizer';
+import { RazorLanguageFeatureBase } from './RazorLanguageFeatureBase';
+import { RazorLanguageServiceClient } from './RazorLanguageServiceClient';
+import { RazorLogger } from './RazorLogger';
+import { LanguageKind } from './RPC/LanguageKind';
+
+export class RazorRenameProvider
+    extends RazorLanguageFeatureBase
+    implements vscode.RenameProvider {
+
+    constructor(
+        documentSynchronizer: RazorDocumentSynchronizer,
+        documentManager: RazorDocumentManager,
+        serviceClient: RazorLanguageServiceClient,
+        logger: RazorLogger) {
+        super(documentSynchronizer, documentManager, serviceClient, logger);
+    }
+
+    public async provideRenameEdits(
+        document: vscode.TextDocument,
+        position: vscode.Position,
+        newName: string,
+        token: vscode.CancellationToken) {
+
+        const projection = await this.getProjection(document, position, token);
+        if (!projection) {
+            return;
+        }
+
+        if (projection.languageKind !== LanguageKind.CSharp) {
+            // We only support C# renames for now.
+            return;
+        }
+
+        const response = await vscode.commands.executeCommand<vscode.WorkspaceEdit>(
+            'vscode.executeDocumentRenameProvider',
+            projection.uri,
+            projection.position,
+            newName,
+        );
+
```

Comment:
oh interesting so we don't need to remap here because we're remapping in the remap C# command? If that's true definitely leave a detailed comment here stating how things flow

Reply:
Yeah. Will add a comment.